docs/rfcs/XXXX_storage_backends.md

```diff
@@ -0,0 +1,248 @@
+If we would like to stop distributing uploaded documents to each study's bucket, we could simply update each study's Document Storage Backend to use the Default Storage Backend. This would result in all uploaded documents going to the application's bucket while all data operations occurring on each individual study's Data Storage Backend.
+
+### Management of External Buckets
+
+Dewrangle will not be responsible for configuration of the external buckets. This responsibility resides with the organization administrator and whoever controls their resources.
+
+## User Flows
+
+### Creating a Storage Backend
+
+Go to organiztion study page and add a new Storage Backend
+
+User provides basic information such as the bucket name and what prefix they want to keep data under
+![](https://i.ibb.co/n6Qxb6Q/Basic.png)
+
+The user will be asked to configure an IAM user within their AWS account that the API may use to interact with the specified bucket.
+![](https://i.ibb.co/ys5xc61/Create-User.png)
+
+The user will need to create a new programmatic IAM user within the AWS dashboard
+
+![](https://i.ibb.co/7bVWVFR/Screenshot-2021-08-10-at-15-47-08-IAM-Management-Console.png)
+
+The policy that is suggested or one that allows for similar access will need to be created to be attached to the new IAM user
+![](https://i.ibb.co/zQhnYT5/Screenshot-2021-08-10-at-15-51-49-IAM-Management-Console.png)
+
+The new IAM user can then be created
+![](https://i.ibb.co/zhj02jx/Screenshot-2021-08-10-at-15-52-42-IAM-Management-Console.png)
+
+In Dewrangle, the user may now paste in the credentials for the new IAM user
+![](https://i.ibb.co/yBXbjTJ/Create-User-1.png)
+
+Once this is done, Dewrangle will confirm that it has the permissions necessary to perform its operations
+
+![](https://i.ibb.co/1fk2B48/Create-User-3.png)
+
+The new storage backend will now be available for use in studies within the organization.
+
+
+### Modifying Storage Backends for Studies
+
+Study administrators may change a study's settings to use any Storage Backend that is in the organization.
+![](https://i.ibb.co/DQZTKCr/Screen-Shot-2021-08-10-at-4-16-15-PM.png)
+
+A study may only have one Document Storage Backend to read and write uploaded documents to. If this location changes after uploads have been made, Dewrangle will not be able to find old files unless they are migrated to the correct location in the new backend and updated in the application database. It may be possible to automate this in the future.
+
+[========]
+
+
+## Technical Implementation
+
+### StorageBackend Model
+
+The storage backend model
+
+```graphql
+type StorageBackend {
+  id: ID!
+  name: String!
+  bucket: String
+  prefix: String
+  organization: Organization
+  accessKey: String
+  secretKey: String
+  studies: [one-many]
```

Comment:
I think this would be many-to-many? Bc a study can have both a document storage backend and data storage backend and those backends may be associated with multiple studies

Reply:
Yeah this may not be accurate. Because studies have both a one-one for document storage and a one-many for data storage, django will put two reverse relations on this model.